pkg/cmd/dev/dev.go

```diff
@@ -36,8 +38,16 @@ func runDev(ctx context.Context, options *DevOptions) error {
 	if err := kitfile.LoadModel(modelfile); err != nil {
 		return err
 	}
-	output.Infof("Loaded Kitfile: %s", kitfile.Model.Path)
-	modelPath, _, err := filesystem.VerifySubpath(options.contextDir, kitfile.Model.Path)
+	output.Infof("Loaded Kitfile: %s", options.modelFile)
+	modelPath := kitfile.Model.Path
+	if kfutils.IsModelKitReference(kitfile.Model.Path) {
+		resolvedKitfile, err := kfutils.ResolveKitfile(ctx, options.configHome, kitfile.Model.Path, kitfile.Model.Path)
+		if err != nil {
+			return fmt.Errorf("Failed to resolve referenced modelkit %s: %w", kitfile.Model.Path, err)
+		}
+		modelPath = resolvedKitfile.Model.Path
```

Comment:
You will also probably want to include the resolved Kitfile's modelparts, if applicable:
```suggestion
		kitfile.Model.Path = resolvedKitfile.Model.Path
		kitfile.Model.Parts = append(kitfile.Model.Parts, resolvedKitfile.Model.Parts...)
```